libs/@guardian/identity-auth-frontend/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { IdentityAuth } from '@guardian/identity-auth';
+import type { CustomClaims } from '@guardian/identity-auth';
+
+type FrontendIdTokenClaims = CustomClaims & {
+	email: string;
+	braze_uuid: string;
+};
+
+export type FrontendIdentityAuth = IdentityAuth<never, FrontendIdTokenClaims>;
+
+const getStage = (isDev: boolean, stage: string) => (isDev ? 'DEV' : stage);
+
+const getIssuer = (stage: string) =>
+	stage === 'PROD'
+		? 'https://profile.theguardian.com/oauth2/aus3xgj525jYQRowl417'
+		: 'https://profile.code.dev-theguardian.com/oauth2/aus3v9gla95Toj0EE0x7';
+
+const getClientId = (stage: string) =>
+	stage === 'PROD' ? '0oa79m1fmgzrtaHc1417' : '0oa53x6k5wGYXOGzm0x7';
+
+const getRedirectUri = (stage: string) => {
+	switch (stage) {
+		case 'PROD':
+			return 'https://www.theguardian.com/';
+		case 'CODE':
+			return 'https://m.code.dev-theguardian.com/';
+		case 'DEV':
+		default:
+			return 'http://localhost:3030/';
+	}
+};
+
+export const getIdentityAuth = () => {
+	const { guardian } = window;
+	const { config } = guardian ?? {};
+	const { isDev, stage } = guardian?.config ?? {};
+
+	if (!guardian || !config || !isDev || !stage) {
+		throw new Error('window.guardian has not yet been initialized');
+	}
```

Comment:
Are you expecting this to throw an error in production, when `isDev === false`?


---

FYI Typescript will happily narrow things for you, so you can go step by step with your errors:

```ts
if(!window.guardian) throw new Error('window.guardian has not yet been initialized');
const { config } = window.guardian 
// `config` is always defined here
// etc…
```

Reply:
> FYI Typescript will happily narrow things for you, so you can go step by step with your errors:

I'm not sure if theres an issue with how Typescript is setup with this project, but typescript was refusing to narrow this for me when I tried!